ts/client/src/accounts/group.ts

```diff
@@ -308,7 +310,7 @@ export class Group {
       await client.program.provider.connection.getMultipleAccountsInfo(oracles);
 
     const coder = new BorshAccountsCoder(client.program.idl);
-    ais.forEach(async (ai, i) => {
+    for await (const [i, ai] of ais.entries()) {
```

Comment:
the await here is key

Reply:
We may be able to make this faster by using Promise.all

Reply:
something like this:

```
const loadPrices = Array.from(ais.entries()).map(async ([i, ai]) => {
      const perpMarket = perpMarkets[i];
      if (!ai)
        throw new Error(
          `Undefined ai object in reloadPerpMarketOraclePrices for ${perpMarket.oracle}!`,
        );
      const { price, uiPrice } = await this.decodePriceFromOracleAi(
        coder,
        perpMarket.oracle,
        ai,
        perpMarket.baseDecimals,
        client,
      );
      perpMarket._price = price;
      perpMarket._uiPrice = uiPrice;
    });

    await Promise.all(loadPrices);
    ```

Reply:
thanks, fixed